dapr/proto/runtime/v1/dapr.proto

```diff
@@ -132,6 +141,9 @@ message SaveStateRequest {
 
   // The array of the state key values.
   repeated common.v1.StateItem states = 2;
+
+  // The metadata which will be sent to state store components.
+  map<string,string> metadata = 3;

```

Comment:
metadata at this level is not needed as StateItem has metadata, adding it here would be confusing.
When you implement [this issue about get, save parity](https://github.com/dapr/dapr/issues/1891), SaveState will become SaveBulkState at that time we can discuss if metadata needs to be moved at api level

Reply:
sounds good. Done .